primitives/runtime/src/traits.rs

```diff
@@ -1210,13 +1213,29 @@ pub trait Header: Clone + Send + Sync + Codec + Eq + MaybeSerialize + Debug + 's
 	}
 }
 
+/// Something that provides the Header Type.
+pub trait HeaderProvider {
+	/// Header type.
+	type HeaderT: Header<Hash = Self::HashT>;
+	/// Block hash type.
+	type HashT: HashOutput;
+}
+
 /// Something which fulfills the abstract idea of a Substrate block. It has types for
 /// `Extrinsic` pieces of information as well as a `Header`.
 ///
 /// You can get an iterator over each of the `extrinsics` and retrieve the `header`.
-pub trait Block: Clone + Send + Sync + Codec + Eq + MaybeSerialize + Debug + 'static {
+pub trait Block: HeaderProvider<HeaderT = <Self as Block>::Header, HashT = <Self as Block>::Hash> + Clone + Send + Sync + Codec + Eq + MaybeSerialize + Debug + 'static {
 	/// Type for extrinsics.
-	type Extrinsic: Member + Codec + Extrinsic + MaybeSerialize;
+	type Extrinsic: Debug
```

Comment:
This change is required? Otherwise it still fails?

Reply:
Will check again.

Reply:
Not needed anymore. Removed.